frontend/src/NavBar.js

```diff
@@ -1,59 +1,42 @@
-import React, { Component } from 'react'
+import React, { useEffect, useState } from 'react'
 import { Link } from 'react-router-dom'
 import SearchBar from './SearchBar'
 
 /**
  * The navigation bar at the top of the page, containing the logo, search bar, and cart icon.
  */
-class NavBar extends Component {
-  constructor(props) {
-    super(props)
 
-    this.state = {
-      courseCount: Object.keys(localStorage).filter(a => !a.startsWith('meta-'))
-        .length,
+export const NavBar = () => {
+  const getCourseCount = () =>
+    Object.keys(localStorage).filter(a => !a.startsWith('meta-')).length
```

Comment:
would be nice to have all localStorage operations be in a helper file called `store` where we can have `getSavedCourses` or something along those lines. Here's a TS rendition I have from a different project:

```ts
/**
 * Keys used in local storage
 */

const ACCESS_TOKEN_KEY = 'accessToken'
const REFRESH_TOKEN_KEY = 'refreshToken'

/**
 * Types relating to sessions
 */

interface IToken {
  email: string
  exp: number
  id: number
  type: EAccountType
}

let cachedStore: Storage

/**
 * Helper functions
 */

const getStore = (): Storage | undefined => {
  if (cachedStore) return cachedStore
  if (!isOnBrowser()) return undefined
  if (typeof window === 'undefined') return undefined
  if (!window || !window.localStorage) return undefined
  cachedStore = window.localStorage
  return cachedStore
}

/**
 * Getter functions
 */

export const getAccessToken = (): string | undefined => {
  const store = getStore()
  return (store && store.getItem(ACCESS_TOKEN_KEY)) || undefined
}

export const getRefreshToken = (): string | undefined => {
  const store = getStore()
  return (store && store.getItem(REFRESH_TOKEN_KEY)) || undefined
}

const getAccessTokenDecoded = (): IToken | undefined => {
  const accessToken = getAccessToken()
  return (accessToken && (jwt.decode(accessToken) as IToken)) || undefined
}

const getRefreshTokenDecoded = (): IToken | undefined => {
  const refreshToken = getRefreshToken()
  return (refreshToken && (jwt.decode(refreshToken) as IToken)) || undefined
}
```

Reply:
If this is a path you want to take, this could be a different PR, as well

Reply:
we're using localStorage for a lot of different things, so this would actually be really nice! I'll add a story for this.